tests/integration_tests/int_test_git_magic.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import os
+import shutil
+import sys
+import tempfile
+import unittest
+from unittest.mock import Mock, patch
+
+from git import Repo
+
+from colony import branch_utils, shell
+from colony.constants import UNCOMMITTED_BRANCH_NAME
+from tests.helpers.repo_utils import (
+    achieve_dirty_and_untracked_repo,
+    add_untracked,
+    create_clean_repo,
+    make_repo_dirty,
+    readonly_handler,
+)
+
+logging.getLogger("git").setLevel(logging.WARNING)
+logger = logging.getLogger(__name__)
+
+
+class GitMagicTests(unittest.TestCase):
```

Comment:
lets rename this class and file to something like: "int_test_temp_branch_with_local_changes" or "int_test_temp_branch" for short

Reply:
ok